src/app/containers/Toggle/useToggle.test.jsx

```diff
@@ -1,51 +1,119 @@
 /* eslint react/prop-types: 0 */
 import React from 'react';
-import { renderHook } from '@testing-library/react-hooks';
+import fetchMock from 'fetch-mock';
+import { renderHook, act } from '@testing-library/react-hooks';
+import '@testing-library/jest-dom/extend-expect';
 import useToggle from './useToggle';
 import { ToggleContextProvider } from '#contexts/ToggleContext';
 
-jest.mock('react', () => {
-  const original = jest.requireActual('react');
-  return {
-    ...original,
-    useContext: jest.fn(),
-  };
-});
+process.env.SIMORGH_APP_ENV = 'test';
+process.env.SIMORGH_TOGGLES_URL = 'https://mock-toggles-endpoint.bbc.co.uk';
 
-const wrapper = ({ children }) => (
-  <ToggleContextProvider>{children}</ToggleContextProvider>
-);
+const togglesUrl = `https://mock-toggles-endpoint.bbc.co.uk/toggles?application=simorgh&service=mundo&__amp_source_origin=https://www.test.bbc.com&geoiplookup=true`;
 
-const { useContext } = jest.requireMock('react');
+afterEach(() => {
+  fetchMock.restore();
+});
 
 describe('useToggle custom hook', () => {
-  afterEach(() => jest.clearAllMocks());
-  test('should return true for foo toggle on test', () => {
-    useContext.mockReturnValueOnce({ env: 'test' }).mockReturnValueOnce({
-      toggleState: {
-        test: { foo: { enabled: true } },
-        live: { foo: { enabled: false } },
-      },
+  describe('Given ads toggle that is fetched from the toggle service is enabled', () => {
+    beforeEach(() => {
+      fetchMock.mock(togglesUrl, {
+        toggles: {
+          ads: {
+            enabled: true,
+            value: '',
+          },
+        },
+      });
     });
 
-    const { result } = renderHook(() => useToggle('foo'), {
-      wrapper,
+    it('return enabled true', async () => {
+      let result;
+      const wrapper = ({ children }) => (
```

Comment:
It would be helpful if we had in our docs links to high standard examples that explain how we set up and test certain things like this example for testing custom hooks. The reason being that I struggle to keep these things in memory but I do remember that I saw a good example somewhere in the codebase and have to spend some time hunting it down. It would be good to do this for lots of things and not just testing approaches.